src/main/java/DataStore/DataStore.java

```diff
@@ -1,9 +1,11 @@
 public interface DataStore {
 
+  public Boolean isDirectory(String uri);
+  public Boolean isFile(String uri);
   public String[] listContent();
   public Boolean existsInStore(String uri);
   public byte[] readFile(String uri);
   public String getFileType(String uri);
   public void postFile (String uri, byte[] content);
```

Comment:
The name `postFile` is a bit unclear. Is it creating a file?

Reply:
It is creating a file, so I've changed the name to `#createFileWithContent`.